src/messaging/ExchangeContext.cpp

```diff
@@ -183,10 +183,17 @@ CHIP_ERROR ExchangeContext::SendMessage(Protocols::Id protocolId, uint8_t msgTyp
             return CHIP_ERROR_INTERNAL;
         }
 
+        //
+        // Our session should still be valid, since it had been evicted, all exchanges on that session would have been torn down
+        // as a side-effect. So for us to get here to send a message and still not have a session, something terribly wrong must
+        // have happened.
+        //
+        VerifyOrDie(mSession);
```

Comment:
I don't think that's a reasonable thing to do.  In particular, if the session was evicted in response to a message, and now we are trying to send a response to that message, that's a slightly silly thing to do, but does not seem to be "worth crashing on" silly...

Reply:
Sure - returning an error now instead.